pkg/cmd/cli/nodeagent/server.go

```diff
@@ -478,3 +488,65 @@ func (s *nodeAgentServer) markInProgressPVRsFailed(client ctrlclient.Client) {
 		s.logger.WithField("podvolumerestore", pvr.GetName()).Warn(pvr.Status.Message)
 	}
 }
+
+var getConfigsFunc = nodeagent.GetConfigs
+
+func (s *nodeAgentServer) getDataPathConcurrentNum(defaultNum int) int {
+	configs, err := getConfigsFunc(s.ctx, s.namespace, s.kubeClient)
+	if err != nil {
+		s.logger.WithError(err).Warn("Failed to get node agent configs")
+		return defaultNum
+	}
+
+	if configs == nil || configs.DataPathConcurrency == nil {
+		s.logger.Infof("Node agent configs are not found, use the default number %v", defaultNum)
+		return defaultNum
+	}
+
+	globalNum := configs.DataPathConcurrency.GlobalConfig
+
+	if globalNum <= 0 {
```

Comment:
Here it is preparing the globalNum only. The number is not being used here --- it will be used later in any case that the perNode number is not available/valid.
So it is fine to have this preparation first and it is convenient for coding since we can restore the prepared globalNum any time as a failback.

Moreover, there indicates one point --- users must set a globalNum first even if they set a perNode number for each node, otherwise, there will be a warning `Global number %v is invalid, use the default value %v`.
I don't think this matters since it is very seldom/inconvenient that setting a perNode number for each node and leave the globalNum empty.